src/Kafka/Producer.hs

```diff
@@ -108,6 +114,62 @@ produceMessage kp@(KafkaProducer (Kafka k) _ (TopicConf tc)) m = liftIO $ do
                 keyPtr (fromIntegral keyLength) nullPtr
 
 
+-- | Sends a single message synchronously
+--
+--   This function implements a simple synchronous production of a message by
+--   passing along an error pointer. Once this pointer has been set to an error
+--   the message can be considered sent. If the message is sent with
+--   'RdKafkaRespErrNoError', then this returns a `Nothing`
+--
+--   Example C-solution used: https://github.com/edenhill/librdkafka/wiki/Sync-producer
+produceMessageSync :: MonadIO m
+                   => KafkaProducer
+                   -> ProducerRecord
+                   -> m (Maybe KafkaError)
```

Comment:
Can we use `m (Either KafkaError ())` instead? I think we have `m (Maybe KafkaError)` pattern and I think it was my mistake to introduce it :(

Reply:
Yeah NP, that makes the test code simpler too